fish-rust/src/wchar_ffi.rs

```diff
@@ -153,3 +153,20 @@ impl WCharFromFFI<Vec<u8>> for cxx::SharedPtr<cxx::CxxString> {
         self.as_bytes().to_vec()
     }
 }
+
+/// Convert from FFI types to a reference to a wide string (i.e. a [`wstr`]) without allocating.
+pub trait AsWstr<'a> {
+    fn as_wstr(&'a self) -> &'a wstr;
+}
+
+impl<'a> AsWstr<'a> for cxx::UniquePtr<cxx::CxxWString> {
+    fn as_wstr(&'a self) -> &'a wstr {
+        wstr::from_char_slice(self.as_chars())
+    }
```

Comment:
nice, I could have used that a bunch of times.
FWIW the lifetime annotation is redundant I think.

Reply:
Yeah, I think the lifetime depends on what version of rustc you're targeting. There's an argument to be made that the excessive from_ffi() to_ffi() calls don't matter since it's just during the transition period, but I still felt ugly enough using them that I thought it was worth my time to add the trait.